// baconfort-react/src/utils/debugUseEffect.js
// Utility para detectar bucles infinitos en useEffect
let effectExecutions = new Map();

export const debugUseEffect = (effectName, dependencies = []) => {
  const key = `${effectName}-${JSON.stringify(dependencies)}`;
  const now = Date.now();
  
  if (!effectExecutions.has(key)) {
    effectExecutions.set(key, []);
  }
  
  const executions = effectExecutions.get(key);
  executions.push(now);
  
  // Mantener solo las 칰ltimas 10 ejecuciones
  if (executions.length > 10) {
    executions.shift();
  }
  
  // Detectar ejecuciones muy r치pidas (m치s de 5 en 1 segundo)
  const oneSecondAgo = now - 1000;
  const recentExecutions = executions.filter(time => time > oneSecondAgo);
  
  if (recentExecutions.length >= 5) {
    console.error(`游뚿 INFINITE LOOP DETECTED: ${effectName} ejecutado ${recentExecutions.length} veces en el 칰ltimo segundo!`);
    console.error('Dependencias:', dependencies);
    console.error('Tiempos de ejecuci칩n recientes:', recentExecutions.map(time => new Date(time).toLocaleTimeString()));
    
    // Limpiar el mapa para evitar spam de logs
    effectExecutions.delete(key);
    
    return true; // Indica que se detect칩 un bucle
  }
  
  return false; // No hay bucle detectado
};

// Funci칩n para limpiar el monitoreo
export const clearEffectDebug = () => {
  effectExecutions.clear();
};

// Funci칩n para obtener estad칤sticas
export const getEffectStats = () => {
  const stats = {};
  for (const [key, executions] of effectExecutions.entries()) {
    stats[key] = {
      totalExecutions: executions.length,
      lastExecution: new Date(executions[executions.length - 1]).toLocaleTimeString(),
      frequency: executions.length > 1 ? 
        Math.round((executions[executions.length - 1] - executions[0]) / executions.length) : 0
    };
  }
  return stats;
};

// Hacer disponible en consola para debugging
if (typeof window !== 'undefined') {
  window.getEffectStats = getEffectStats;
  window.clearEffectDebug = clearEffectDebug;
}
